setup.py

```diff
@@ -69,53 +69,31 @@
 
 import os
 import re
-import shutil
-from pathlib import Path
 
 from setuptools import Command, find_packages, setup
 
 
-# Remove stale transformers.egg-info directory to avoid https://github.com/pypa/pip/issues/5466
-stale_egg_info = Path(__file__).parent / "transformers.egg-info"
-if stale_egg_info.exists():
-    print(
-        (
-            "Warning: {} exists.\n\n"
-            "If you recently updated transformers to 3.0 or later, this is expected,\n"
-            "but it may prevent transformers from installing in editable mode.\n\n"
-            "This directory is automatically generated by Python's packaging tools.\n"
-            "I will remove it now.\n\n"
-            "See https://github.com/pypa/pip/issues/5466 for details.\n"
-        ).format(stale_egg_info)
-    )
-    shutil.rmtree(stale_egg_info)
-
-
 # IMPORTANT:
 # 1. all dependencies should be listed here with their version requirements if any
 # 2. once modified, run: `make deps_table_update` to update src/transformers/dependency_versions_table.py
 _deps = [
-    "Pillow>=10.0.1,<=15.0",
     "accelerate>=0.26.0",
     "av",
     "beautifulsoup4",
     "blobfile",
     "codecarbon>=2.8.1",
     "cookiecutter==1.7.3",
-    "dataclasses",
     "datasets!=2.5.0",
     "deepspeed>=0.9.3",
     "diffusers",
-    "dill<0.3.5",
     "evaluate>=0.2.0",
     "faiss-cpu",
     "fastapi",
     "filelock",
     "flax>=0.4.1,<=0.7.0",
-    "fsspec<2023.10.0",
     "ftfy",
     "fugashi>=1.0",
-    "GitPython<3.1.19",
```

Comment:
The pin on GitPython was added to fix CI (https://github.com/huggingface/transformers/pull/12858) -- removing to test whether it is still needed